### PR TITLE
Handle aliases in graphql-flow.

### PR DESCRIPTION
**Pull request by jeresig**
## Summary:
This adds a new option to the config for graphql-flow so that you can specify aliases to map to, to help resolve imports. I used the same format as Vite to keep things simple. This will make it so that we can support our `~/`-style imports.

In the process I realized a number of things about this repo were broken, so I fixed them:

- We weren't running tsc, so I fixed that, and then fixed a bunch of TS failures that were hidden.
- The ESLint config was broken, so I fixed that (and then fixed all the failures that came up).
- Prettier was broken (still expecting Flow), so I fixed that.
- VS Code wasn't set up properly, so I fixed that.
- The Github PR Workflow was broken, still expecting Flow/.js files, so I fixed that.

Hopefully this helps things to run more smoothly!

Issue: FEI-5745

## Test plan:
I ran `yarn tsc` and `yarn eslint src/**/*.ts` and they both passed.

### CODE REVIEW

File: src/parser/utils.ts [hunk 1]
```diff
@@ -1,24 +1,37 @@
-import fs from 'fs';
+import fs from "fs";
+import {Config} from "../types";
 
-export const getPathWithExtension = (pathWithoutExtension: string): string => {
+export const getPathWithExtension = (
+    pathWithoutExtension: string,
+    config: Config,
+): string => {
+    // Process the path so that we can handle aliases.
```
Comment (jeresig) [inline src/parser/utils.ts:8]: This is the new change.

File: src/parser/__test__/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        jest.spyOn(fs, "existsSync").mockImplementation((path) => false);
+
+        // Act
+        const result = getPathWithExtension("/path/to/file", config);
+
+        // Assert
+        expect(result).toBe("");
+    });
+
+    it("maps aliases to their correct value", () => {
```
Comment (jeresig) [inline src/parser/__test__/utils.test.ts:64]: This is the test confirming that it works.

File: src/types.ts [hunk 3]
```diff
@@ -8,90 +8,94 @@ import type {
...
+    ignorePragma?: string;
+    dumpOperations?: string;
 };
 
 export type Config = {
-    crawl: CrawlConfig
-    generate: GenerateConfig | Array<GenerateConfig>
+    crawl: CrawlConfig;
+    generate: GenerateConfig | Array<GenerateConfig>;
+    alias?: Array<{
```
Comment (jeresig) [inline src/types.ts:46]: This is the format of the new alias config setting.

**Review by kevinb-khan — APPROVED** [review]
LGTM.  Did we only recently run into issues with graphql-flow having troubles traversing `~/` style imports?

Thanks for fixing all of the tooling in this repo.  Sorry for missing that during the TypeScript upgrade.

**Comment by jeresig** [comment]
@kevinb-khan So it turns out that I hit this during: https://github.com/Khan/webapp/pull/23926 we were hitting issues where we were importing GraphQL fragments across packages using a `~/` prefix. Before this point there hadn't been any cases of this, but once I started using `~/` everywhere this code started to break.

**Pull request merged**